### PR TITLE
CORS-3546: Nutanix: support for gpus and dataDisks

### PR DESCRIPTION
**Pull request by yanhua121**
Add suport for the feature stories:
[OCPSTRAT-1314](https://issues.redhat.com//browse/OCPSTRAT-1314): Allow creating Nutanix worker VMs with GPUs
[OCPSTRAT-1394](https://issues.redhat.com//browse/OCPSTRAT-1394): Allow creating Nutanix VMs with multiple disks

The api changes are in https://github.com/openshift/api/pull/1935.

Manual in-lab testing of the this PR passed.

### CODE REVIEW

File: pkg/actuators/machine/utils.go [hunk 1]
```diff
@@ -398,3 +403,97 @@ func GetMibValueOfQuantity(quantity resource.Quantity) int64 {
 func Ptr[T any](v T) *T {
 	return &v
 }
+
+// GetGPU returns the VMGpu for the given Prism Element (uuid) and GPU identifier input
+func GetGPU(ctx context.Context, client *nutanixClientV3.Client, peUUID string, gpu machinev1.NutanixGPU) (*nutanixClientV3.VMGpu, error) {
+	peGPUs, err := GetGPUsForPE(ctx, client, peUUID)
+	if err != nil {
+		return nil, err
```
Comment (JoelSpeed) [inline pkg/actuators/machine/utils.go:411]: Would appreciate if we can wrap errors at every level to add additional context. Makes it easier to debug as you can search for the various messages in the final message and find the stack

File: pkg/actuators/machine/utils.go [hunk 2]
```diff
@@ -398,3 +403,97 @@ func GetMibValueOfQuantity(quantity resource.Quantity) int64 {
 func Ptr[T any](v T) *T {
 	return &v
 }
+
+// GetGPU returns the VMGpu for the given Prism Element (uuid) and GPU identifier input
+func GetGPU(ctx context.Context, client *nutanixClientV3.Client, peUUID string, gpu machinev1.NutanixGPU) (*nutanixClientV3.VMGpu, error) {
```
Comment (JoelSpeed) [inline pkg/actuators/machine/utils.go:408]: Does this need to be a public method?

File: pkg/actuators/machine/utils.go [hunk 3]
```diff
@@ -398,3 +403,97 @@ func GetMibValueOfQuantity(quantity resource.Quantity) int64 {
...
+	foundGPU, err := GetGPUFromList(ctx, client, gpu, peGPUs)
+	if err != nil {
+		return nil, err
+	}
+
+	return foundGPU, nil
+}
+
+// GetGPUList returns a list of VMGpus for the given list of GPU identifiers in the Prism Element (uuid)
+func GetGPUList(ctx context.Context, client *nutanixClientV3.Client, gpus []machinev1.NutanixGPU, peUUID string) ([]*nutanixClientV3.VMGpu, error) {
```
Comment (JoelSpeed) [inline pkg/actuators/machine/utils.go:426]: As above, does this really need to be public?

File: pkg/actuators/machine/utils.go (hunk 1 above)
Comment (yanhua121) [reply pkg/actuators/machine/utils.go]: addressed.

File: pkg/actuators/machine/utils.go (hunk 2 above)
Comment (yanhua121) [reply pkg/actuators/machine/utils.go]: Changed the method name to start with lower case.

File: pkg/actuators/machine/utils.go (hunk 3 above)
Comment (yanhua121) [reply pkg/actuators/machine/utils.go]: Changed the method name to start with lower case.

**Review by elmiko — COMMENTED** [review]
i'm not an expert on the nutanix platform, but this PR makes sense to me.
/lgtm

**Pull request merged**